### PR TITLE
Add Tabs component

### PR DESCRIPTION
**Pull request by jonkoops**
See the reference from the Design System: https://designsystem.amsterdam.nl/7awj1hc9f/p/15737c-tab-navigation

### CODE REVIEW

File: packages/asc-ui/src/components/Tabs/Tabs.tsx [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+          const tabId = formatTabId(props.id)
+          const panelId = formatPanelId(props.id)
+
+          return (
+            <TabButton
+              key={props.id}
+              id={tabId}
+              aria-controls={panelId}
+              aria-selected={isSelected}
+              tabIndex={isFocusable ? 0 : -1}
```
Comment (alimpens) [inline packages/asc-ui/src/components/Tabs/Tabs.tsx:111]: tabindex shouldn't change on focus, only on selection. See implementation [here](https://www.w3.org/TR/wai-aria-practices-1.1/examples/tabs/tabs-2/tabs.html). The problem with changing tabindex on focus is that when you leave the tab list and return, the focus also returns to the last focussed tab, not the selected tab. This doesn't match with _When focus moves into the tab list, places focus on the active tab element_ from that same link.

File: packages/asc-ui/src/components/Tabs/Tabs.tsx [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+  const focusPreviousTab = () => focusTowards(FocusDirection.Backward)
+
+  function handleKeyDown(event: React.KeyboardEvent<HTMLDivElement>) {
+    if (event.key === 'ArrowRight') {
+      focusNextTab()
+    }
+
+    if (event.key === 'ArrowLeft') {
+      focusPreviousTab()
+    }
```
Comment (alimpens) [inline packages/asc-ui/src/components/Tabs/Tabs.tsx:93]: Do we also want to provide Home- and End-button functionality? You can find an example of that [here](https://www.w3.org/TR/wai-aria-practices-1.1/examples/tabs/tabs-2/tabs.html).

Comment (chrisvanmook) [reply packages/asc-ui/src/components/Tabs/Tabs.tsx:93]: We actually have a hook for this that we can use here: `useFocusWithArrows`, which includes the Home and End buttons

File: packages/asc-ui/src/utils/hooks/useFocusWithArrows.tsx [hunk 3]
```diff
@@ -8,12 +8,16 @@ import { FOCUSABLE_ELEMENTS } from './constants'
  * @param ref Component ref
  * @param rotating Jump to first item from last or vice versa
  * @param directChildrenOnly Useful if you don't want to focus on other focussable elements inside the child components of the ref
+ * @param horizontally In case you need to navigate horizontally, using left / right arrow buttons
  */
 const useFocusWithArrows = (
   ref: React.RefObject<HTMLElement>,
   rotating = false,
   directChildrenOnly = false,
+  horizontally = false,
```
Comment (chrisvanmook) [inline packages/asc-ui/src/utils/hooks/useFocusWithArrows.tsx:17]: Probably need to refactor this to accept an object with options, now we need to pass 4 arguments

Comment (jonkoops) [reply packages/asc-ui/src/utils/hooks/useFocusWithArrows.tsx:17]: Yeah, agreed this is a tad confusing right now.

File: stories/src/ui/Tabs.stories.mdx [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+      </Tab>
+      <Tab id="two" label="Second">
+        Contents of the second tab.
+      </Tab>
+      <Tab id="three" label="Last">
+        Contents of the last tab.
+      </Tab>
+    </Tabs>
+  </Story>
+</Preview>
```
Comment (chrisvanmook) [inline stories/src/ui/Tabs.stories.mdx:28]: I saw that it also works well on mobile with the horizontal scroll, nice! Should we add a mobile story here as well just for convenience? You can use a storybook addon for this

File: packages/asc-ui/src/components/Tabs/TabButtonStyle.ts [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+type Props = {
+  isSelected?: boolean
+}
+
+const TabButtonStyle = styled.button<Props>`
+  background-color: transparent;
+  padding: 12px 15px;
+  color: ${themeColor('tint', 'level7')};
+  border: none;
+  margin-right: 30px;
```
[the remark this answers is not in the document]
Comment (jonkoops) [reply packages/asc-ui/src/components/Tabs/TabButtonStyle.ts]: Fixed

File: packages/asc-ui/src/components/Tabs/TabButtonStyle.ts [hunk 6]
```diff
@@ -0,0 +1,35 @@
+import styled, { css } from 'styled-components'
+import { themeColor } from '../../utils'
+
+type Props = {
+  isSelected?: boolean
+}
+
+const TabButtonStyle = styled.button<Props>`
+  background-color: transparent;
+  padding: 12px 15px;
```
[the remark this answers is not in the document]
Comment (jonkoops) [reply packages/asc-ui/src/components/Tabs/TabButtonStyle.ts]: Fixed

**Review by alimpens — APPROVED** [review]

File: packages/asc-ui/src/components/Tabs/TabList.tsx [hunk 7]
```diff
@@ -1,45 +1,26 @@
...
-  useImperativeHandle,
-  useRef,
-} from 'react'
-import TabListStyle, { TabListWrapper, TabListFade } from './TabListStyle'
+import styled from 'styled-components'
+import { themeColor, themeSpacing } from '../../utils'
 
-export default forwardRef<HTMLDivElement, HTMLAttributes<HTMLDivElement>>(
-  ({ children, ...otherProps }, externalRef) => {
-    const [canScroll, setCanScroll] = useState<boolean>(false)
```
Comment (alimpens) [inline packages/asc-ui/src/components/Tabs/TabList.tsx]: Without checking if TabList can scroll you always have a small fade on the right, which you can see over the inset box shadow. It's quite small, but it can imply for some users that TabList is scrollable when it isn't. Should we keep canScroll here?

File: packages/asc-ui/src/components/Tabs/Tabs.tsx [hunk 8]
```diff
@@ -33,7 +33,7 @@ function formatPanelId(id: string) {
   return `panel-${id}`
 }
 
-export const Tabs: React.FC<Tabs> = ({ label, children, ...otherProps }) => {
```
Comment (alimpens) [inline packages/asc-ui/src/components/Tabs/Tabs.tsx]: Why did you remove `otherProps` here? The idea was that the user can set props (such as `style`) on the `Tabs` component, in order to style `TabList `for example.

File: packages/asc-ui/src/components/Tabs/Tabs.tsx [hunk 9]
```diff
@@ -43,38 +43,28 @@ export const Tabs: React.FC<Tabs> = ({ label, children, ...otherProps }) => {
...
   return (
     <>
-      <TabList
-        aria-label={label}
-        onKeyDown={keyDown}
-        ref={tabListRef}
-        {...otherProps}
-      >
-        {children.map(
-          ({ props: { id, label: childLabel, ...otherChildProps } }) => {
```
Comment (alimpens) [inline packages/asc-ui/src/components/Tabs/Tabs.tsx]: Why did you refactor this? Destructuring the props is more readable, in my opinion. And why did you remove `otherChildProps`? The idea was to enable the user to set props (such as `style`) on `Tab`, in order to style `TabButton`.

File: packages/asc-ui/src/components/Tabs/Tabs.tsx (hunk 1 above)
Comment (alimpens) [reply packages/asc-ui/src/components/Tabs/Tabs.tsx:59]: This behaviour seems to be broken again, the focus seems to return to the last focussed tab, not the selected tab. I can't quite figure out why this changed, though :)

**Review by alimpens — COMMENTED** [review]
This commit depends on the answer to [this comment](https://github.com/Amsterdam/amsterdam-styled-components/pull/975/commits/f0679515bbab4d15ad7e567e65a2d706e5b30277#r473926714), I think.

**Comment by timvanoostrom** [comment]
Also, I've compared the design with the one referenced in designsystem. 
Screen Shot 2020-08-24 at 09 21 44
Screen Shot 2020-08-24 at 09 22 43

A couple of things stand out:

- there is no padding in the tab buttons in the design reference
- the font seems to be larger and possibly not the same weight

**Comment by timvanoostrom** [comment]
I see the font specs are written out so maybe the design or the specs are wrong. I guess some clarification is needed.

File: packages/asc-ui/src/components/Tabs/Tabs.tsx [hunk 10]
```diff
@@ -0,0 +1,96 @@
...
+const TabPanel = styled.div.attrs({
+  role: 'tabpanel',
+  tabIndex: 0,
+})``
+
+export interface TabsProps {
+  /**
+   * A descriptive label for the tabs, required for accessibility.
+   */
+  label: string
```
[the remark this answers is not in the document]
Comment (alimpens) [reply packages/asc-ui/src/components/Tabs/Tabs.tsx:26]: This `label` is used to set `aria-label`, you can't use ReactNode there. I assume you mean the `label` in Tab.tsx?

[the remark this answers is not in the document]
Comment (timvanoostrom) [reply packages/asc-ui/src/components/Tabs/Tabs.tsx:26]: Ah, It's actually used for both. So maybe we should make distinction. 

```
label: ReactNode
ariaLabel: string;
```

[the remark this answers is not in the document]
Comment (timvanoostrom) [reply packages/asc-ui/src/components/Tabs/Tabs.tsx:26]: Ah I see, then it applies to: https://github.com/Amsterdam/amsterdam-styled-components/pull/975/files/5b124f88b20580a137a9b0ac1c3a364e9b46b6d2#diff-c14a52293bc086d98059a5983e3571a8R9

File: packages/asc-ui/src/components/Tabs/Tab.tsx [hunk 11]
```diff
@@ -0,0 +1,14 @@
+import React, { HTMLAttributes } from 'react'
+
+// Some props are omitted since they cannot be overwritten.
+type OmittedProps = 'role' | 'aria-controls' | 'aria-selected' | 'tabIndex'
+
+export interface TabProps
+  extends Omit<HTMLAttributes<HTMLButtonElement>, OmittedProps> {
+  id: string
+  label: string
```
Comment (timvanoostrom) [inline packages/asc-ui/src/components/Tabs/Tab.tsx:9]: You could make this `ReactNode` so that some other elements than `string` can be passed. For example an `<Icon />`.

File: packages/asc-ui/src/components/Tabs/Tabs.tsx (hunk 10 above)
[the remark this answers is not in the document]
Comment (timvanoostrom) [reply packages/asc-ui/src/components/Tabs/Tabs.tsx:26]: Does a component like this need an `aria-label` at all ? It's a `role="tablist"` right?

https://www.marcozehe.de/advanced-aria-tip-1-tabs-in-web-apps/

[the remark this answers is not in the document]
Comment (timvanoostrom) [reply packages/asc-ui/src/components/Tabs/Tabs.tsx:26]: Nah, W3 says `Provides a label that describes the purpose of the set of tabs.` so all good.

**Comment by timvanoostrom** [comment]
LGTM!

**Review by chrisvanmook — APPROVED** [review]

File: packages/asc-ui/src/components/Tabs/TabButton.tsx [hunk 12]
```diff
@@ -5,10 +5,10 @@ const TabButton = styled.button.attrs({
   role: 'tab',
 })<{ isSelected?: boolean }>`
   background-color: transparent;
-  padding: ${themeSpacing(3, 4)};
+  padding: ${themeSpacing(3, 0)};
   color: ${themeColor('tint', 'level7')};
   border: none;
-  margin-right: ${themeSpacing(8)};
+  margin-right: ${themeSpacing(7)};
```
Comment (jonkoops) [inline packages/asc-ui/src/components/Tabs/TabButton.tsx:11]: @alimpens Be aware that the fade should also be changed if this is changed.
